src/Microsoft.OData.Core/Batch/ODataBatchWriter.cs

```diff
@@ -383,12 +385,14 @@ public void Flush()
 
         /// <summary>Flushes the write buffer to the underlying stream asynchronously.</summary>
         /// <returns>A task instance that represents the asynchronous operation.</returns>
-        public Task FlushAsync()
+        public async Task FlushAsync()
         {
             this.VerifyCanFlush(false);
 
             // Make sure we switch to writer state Error if an exception is thrown during flushing.
-            return this.FlushAsynchronously().FollowOnFaultWith(t => this.SetState(BatchWriterState.Error));
+            await this.FlushAsynchronously()
+                .FollowOnFaultWith(t => this.SetState(BatchWriterState.Error))
+                .ConfigureAwait(false);

```

Comment:
What does `FollowOnFaulWith` do? Could this be safely transformed to:

```c#
try {
    await this.FlushAsynchronously();
} catch {
    this.SetState(BatchWriterState.Error);
}
```

Reply:
`FollowOnFaultWith` is designed such that the `Action` delegate is invoked only when the antecedent task fails (i.e. task is in a Faulted state)

Reply:
It could be transformed they way you demonstrated. But instead of littering the code with such try...catch I think it's neater to encapsulate that in a helper method